src/main/kotlin/MenuScreen.kt

```diff
@@ -0,0 +1,205 @@
+                val note = notes[noteIndex]
+                println("Заметка '$note' выбрана.")
+                interactWithNote(note)
+            } else {
+                println("Заметка с номером $noteIndex не найдена.")
+            }
+        } else {
+            println("Некорректный ввод. Пожалуйста, введите число.")
+        }
+    }
+
+    private fun interactWithNote(note: Note) {
+        while (true) {
+            println("Меню взаимодействия с заметкой:")
+            println("1. Редактировать название заметки")
+            println("2. Редактировать содержимое заметки")
+            println("3. Удалить заметку")
+            println("4. Вернуться к списку заметок")
+            print("Выберите пункт меню: ")
+
+            val userInput = readIntFromUser()
+            when (userInput) {
+                1 -> editNoteTitle(note)
+                2 -> editNoteContent(note)
+                3 -> deleteNoteFromArchive(note)
+                4 -> return
+                else -> println("Некорректный ввод. Пожалуйста, выберите существующий пункт.")
+            }
+        }
+    }
+
+    private fun editNoteTitle(note: Note) {
+        print("Введите новое название заметки: ")
+        val newTitle = scanner.nextLine()
+        note.title = newTitle
+        println("Название заметки изменено.")
+    }
+
+    private fun editNoteContent(note: Note) {
+        print("Введите новое содержимое заметки: ")
+        val newContent = scanner.nextLine()
+        note.content = newContent
+        println("Содержимое заметки изменено.")
+    }
+
+    private fun deleteNoteFromArchive(note: Note) {
+        val archive = archivesManager.getArchiveContainingNote(note)
+        if (archive != null) {
+            archive.deleteNoteFromArchive(note)
+            println("Заметка удалена из архива.")
+        } else {
+            println("Не удалось найти архив, содержащий данную заметку.")
+        }
+    }
+
+    private fun interactWithArchives() {
+        val archives = archivesManager.getArchives()
+        if (archives.isEmpty()) {
+            println("Список архивов пуст.")
+            return
+        }
+
+        println("Список архивов:")
+        for ((index, archive) in archives.withIndex()) {
+            println("$index. Архив ${archive.name}")
+        }
+
+        print("Выберите номер архива для выполнения операции (для выхода введите 'exit'): ")
+
+        val userInput = scanner.nextLine()
+        if (userInput.toLowerCase() == "exit") {
+            return
+        }
+
+        val archiveIndex = userInput.toIntOrNull()
+        if (archiveIndex != null) {
+            val archives = archivesManager.getArchives()
+            if (archiveIndex in archives.indices) {
+                val archive = archives[archiveIndex]
+                println("Вы выбрали архив '{archive.name}'.")
+                viewArchiveNotes(archive)
+            } else {
+                println("Архив с номером archiveIndex не найден.")
```

Comment:
```suggestion
                println("Архив с номером $archiveIndex не найден.")
```
а то сейчас вместо индекса выводится строка `archiveIndex `